autoload/vista/parser/lsp.vim

```diff
@@ -41,80 +43,84 @@ function! s:IsFileUri(uri) abort
   return stridx(a:uri, 'file:///') == 0
 endfunction
 
-" The kind field in the result is a number instead of a readable text, we
-" should transform the number to the symbol text first.
-function! vista#parser#lsp#KindToSymbol(line, container) abort
-  let line = a:line
-  " SymbolInformation interface
-  if has_key(line, 'location')
-    let location = line.location
-    if s:IsFileUri(location.uri)
-      let lnum = location.range.start.line + 1
-      let col = location.range.start.character + 1
-      call add(a:container, {
-         \ 'lnum': lnum,
-         \ 'col': col,
-         \ 'kind': s:Kind2Symbol(line.kind),
-         \ 'text': line.name,
-         \ })
-    endif
-  " DocumentSymbol class
-  elseif has_key(line, 'range')
-    let range = line.range
-    let lnum = range.start.line + 1
-    let col = range.start.character + 1
-    call add(a:container, {
-          \ 'lnum': lnum,
-          \ 'col': col,
-          \ 'kind': s:Kind2Symbol(line.kind),
-          \ 'text': line.name,
-          \ })
-    if has_key(line, 'children')
-      for child in line.children
-        call vista#parser#lsp#KindToSymbol(child, a:container)
-      endfor
-    endif
-  endif
+function! s:LspToLocalSymbol(sym, range)
+  return {
+    \ 'lnum': a:range.start.line + 1,
+    \ 'col': a:range.start.character + 1,
+    \ 'kind': s:Kind2Symbol(a:sym.kind),
+    \ 'text': a:sym.name,
+    \ }
 endfunction
 
-function! vista#parser#lsp#CocSymbols(symbol, container) abort
-  if vista#ShouldIgnore(a:symbol.kind)
-    return
-  endif
+function! s:LocalToRawSymbol(sym)
+  return {
+    \ 'line': a:sym.lnum,
+    \ 'kind': a:sym.kind,
+    \ 'name': a:sym.text,
+    \ }
+endfunction
 
-  let raw = { 'line': a:symbol.lnum, 'kind': a:symbol.kind, 'name': a:symbol.text }
-  call add(g:vista.raw, raw)
+function! s:IsDocumentSymbol(sym)
+  return has_key(a:sym, 'selectionRange')
+endfunction
 
-  if a:symbol.kind ==? 'Method' || a:symbol.kind ==? 'Function'
-    call add(g:vista.functions, a:symbol)
-  endif
+function! s:ParseSymbolInfoList(outlist, symbols) abort
+  for lspsym in a:symbols
+    let loc = lspsym.location
+    if s:IsFileUri(loc.uri)
+      call add(a:outlist, s:LspToLocalSymbol(lspsym, loc.range))
+    endif
+  endfor
```

Comment:
Can be simplified as:

```suggestion
let outlist = map(filter(symbols, 's:IsFileUri(v:val["localtion"]["uri"])'), 's:LspToLocalSymbol(v:val, v:val["location"]["range"])')
```

Then no need to define this function.

Reply:
The change is ok, but I'd rather leave the function in place to highlight complementary nature of `ParseSymboInfoList` and `ParseDocumentSymbolsRec`. Actually both of them use the same functional interface for consistency.